src/settings.cpp

```diff
@@ -445,6 +445,17 @@ vstrCentralServerAddress[0] = GetIniSetting ( IniXMLDocument, "client", "central
         eCentralServerAddressType = AT_DEFAULT;
     }
 
+    // custom directory server index
+    if ( GetNumericIniSet ( IniXMLDocument, "client", "customdirectoryindex", 0, MAX_NUM_SERVER_ADDR_ITEMS, iValue ) )
+    {
+        iCustomDirectoryIndex = iValue;
+    }
+    else
+    {
+        // if no custom directory index is found in the settings .ini file, then initialize to zero
```

Comment:
"Note: this is the correct value if `eCentralServerAddressType` is currently `AT_CUSTOM`."

(And if it isn't, set `eCentralServerAddressType` is currently `AT_DEFAULT`...)

Reply:
I think I understand your point here - a manual edit to the config file could leave it with an invalid combination of values.  I think my latest (final?) commit takes care of it.